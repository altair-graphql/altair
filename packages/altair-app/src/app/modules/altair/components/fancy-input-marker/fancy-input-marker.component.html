<mark [ngClass]="{ 'fancy-input-marker__invalid': !resolvedValue }">{{
  section.content
}}</mark>
@if (resolvedValue) {
  <div class="value-indicator__wrapper">
    .
    <div
      nz-tooltip
      [nzTooltipTitle]="resolvedValue"
      class="value-indicator"
    ></div>
  </div>
}
