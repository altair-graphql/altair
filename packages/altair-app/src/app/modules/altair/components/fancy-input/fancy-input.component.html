<div class="fancy-input-container">
  <input
    #fancyInputEl
    type="text"
    class="fancy-input-content fancy-input-element"
    [placeholder]="placeholder"
    [(ngModel)]="value"
    (input)="handleInput()"
    (scroll)="handleScroll()"
    (keydown)="handleKeydown()"
    (blur)="handleBlur()"
    (keydown.enter)="handleSubmit()"
    />
  <div class="fancy-input-backdrop">
    <div
      class="fancy-input-content fancy-input-highlights"
      #fancyInputHighlightsEl
      >
      @for (section of highlightData.sections; track trackByIndex($index, section)) {
        @if (!section.type) {
          <span>{{ section.content }}</span>
        }
        @if (section.type === 'mark') {
          <app-fancy-input-marker
            [section]="section"
            [activeEnvironment]="activeEnvironment"
            (click)="fancyInputEl.focus()"
          ></app-fancy-input-marker>
        }
      }
    </div>
  </div>
</div>
