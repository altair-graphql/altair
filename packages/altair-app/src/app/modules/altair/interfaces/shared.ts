export interface IDictionary<V = any> {
  [key: string]: V;
}

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export type TODO = any;

export type UnknownError = TODO;

export interface TrackByIdItem {
  id: string;
}
