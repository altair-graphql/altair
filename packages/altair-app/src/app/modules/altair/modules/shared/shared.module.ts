import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';

import { DragDropModule } from '@angular/cdk/drag-drop';
import { TranslateModule } from '@ngx-translate/core';
import { NgPipesModule } from 'ngx-pipes';
import { NguiAutoCompleteModule } from '@ngui/auto-complete';
import { MarkdownModule } from 'ngx-markdown';
import { HotToastModule } from '@ngneat/hot-toast';
import { ResizableModule } from 'angular-resizable-element';

import { NzButtonModule } from 'ng-zorro-antd/button';
import { NzFormModule } from 'ng-zorro-antd/form';
import { NzSelectModule } from 'ng-zorro-antd/select';
import { NzModalModule } from 'ng-zorro-antd/modal';
import { NzDropDownModule } from 'ng-zorro-antd/dropdown';
import { NzSwitchModule } from 'ng-zorro-antd/switch';
import { NzListModule } from 'ng-zorro-antd/list';
import { NzRadioModule } from 'ng-zorro-antd/radio';
import { NzAlertModule } from 'ng-zorro-antd/alert';
import { NzLayoutModule } from 'ng-zorro-antd/layout';
import { NzMenuModule } from 'ng-zorro-antd/menu';
import { NzCollapseModule } from 'ng-zorro-antd/collapse';
import { NzInputModule } from 'ng-zorro-antd/input';
import { NzInputNumberModule } from 'ng-zorro-antd/input-number';
import { NzAutocompleteModule } from 'ng-zorro-antd/auto-complete';
import { NzCheckboxModule } from 'ng-zorro-antd/checkbox';
import { NzToolTipModule } from 'ng-zorro-antd/tooltip';
import { NzTabsModule } from 'ng-zorro-antd/tabs';
import { NzTableModule } from 'ng-zorro-antd/table';
import { NzTreeSelectModule } from 'ng-zorro-antd/tree-select';
import { NzGridModule } from 'ng-zorro-antd/grid';
import { NzPopoverModule } from 'ng-zorro-antd/popover';

import { IconsModule } from '../icons/icons.module';
import { NzAvatarModule } from 'ng-zorro-antd/avatar';

const AntdComponentModules = [
  NzButtonModule,
  NzInputModule,
  NzInputNumberModule,
  NzAutocompleteModule,
  NzCheckboxModule,
  NzFormModule,
  NzGridModule,
  NzTabsModule,
  NzSelectModule,
  NzModalModule,
  NzDropDownModule,
  NzSwitchModule,
  NzListModule,
  NzRadioModule,
  NzAlertModule,
  NzLayoutModule,
  NzMenuModule,
  NzCollapseModule,
  NzToolTipModule,
  NzTableModule,
  NzTreeSelectModule,
  NzPopoverModule,
  NzAvatarModule,
  // NgZorroAntdModule,
];

@NgModule({
  imports: [
    CommonModule,
    DragDropModule,
    TranslateModule.forRoot(),
    NgPipesModule,
    NguiAutoCompleteModule, // use antd
    MarkdownModule.forRoot(),
    ResizableModule,
    IconsModule,
    // ToastrModule.forRoot({
    //   newestOnTop: false,
    //   closeButton: true,
    //   positionClass: 'toast-top-center',
    //   enableHtml: true,
    //   countDuplicates: true,
    //   preventDuplicates: true,
    //   resetTimeoutOnDuplicate: true,
    // }),
    // HotToastModule.forRoot({
    //   position: 'top-center',
    //   reverseOrder: true,
    //   dismissible: true,
    //   closeStyle: {
    //     backgroundImage: `none`,
    //   },
    // }),
    ...AntdComponentModules,
  ],

  // DON'T FORGET TO EXPORT MODULE
  exports: [
    DragDropModule,
    TranslateModule,
    NgPipesModule,
    NguiAutoCompleteModule,
    MarkdownModule,
    ResizableModule,
    IconsModule,
    HotToastModule,
    ...AntdComponentModules,
  ],
})
export class SharedModule {
  static forRoot(): ModuleWithProviders<SharedModule> {
    return {
      ngModule: SharedModule,
      providers: [],
    };
  }
}
