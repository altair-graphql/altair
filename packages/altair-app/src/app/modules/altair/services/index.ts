export { ApiService } from './api/api.service';
export { GqlService } from './gql/gql.service';
export { DbService } from './db.service';
export { WindowService } from './window.service';
export { NotifyService } from './notify/notify.service';
export { DonationService } from './donation.service';
export { ElectronAppService } from './electron-app/electron-app.service';
export { KeybinderService } from './keybinder/keybinder.service';
export { StorageService } from './storage/storage.service';
export { QueryCollectionService } from './query-collection/query-collection.service';
export { EnvironmentService } from './environment/environment.service';
export { PluginRegistryService } from './plugin/plugin-registry.service';
export { PluginEventService } from './plugin/plugin-event.service';
export { PreRequestService } from './pre-request/pre-request.service';
export { ThemeRegistryService } from './theme/theme-registry.service';
export { SubscriptionProviderRegistryService } from './subscriptions/subscription-provider-registry.service';
export { PluginContextService } from './plugin/context/plugin-context.service';
export { QueryService } from './query/query.service';
export { AccountService } from './account/account.service';
