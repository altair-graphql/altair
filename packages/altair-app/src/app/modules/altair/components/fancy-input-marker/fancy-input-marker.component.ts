import {
  ChangeDetectionStrategy,
  Component,
  Input,
  OnChanges,
  SimpleChanges,
} from '@angular/core';
import { IDictionary } from '../../interfaces/shared';
import {
  EnvironmentService,
  VARIABLE_REGEX,
} from '../../services/environment/environment.service';
import { HighlightSection } from '../fancy-input/fancy-input.component';

@Component({
  selector: 'app-fancy-input-marker',
  templateUrl: './fancy-input-marker.component.html',
  styles: [],
  changeDetection: ChangeDetectionStrategy.OnPush,
  preserveWhitespaces: false,
})
export class FancyInputMarkerComponent implements OnChanges {
  @Input() section: HighlightSection = { content: '' };
  @Input() activeEnvironment: IDictionary = {};

  resolvedValue = '';

  constructor(private environmentService: EnvironmentService) {}

  ngOnChanges(changes: SimpleChanges) {
    this.resolvedValue = this.environmentService.hydrate(this.section.content);
  }
}
