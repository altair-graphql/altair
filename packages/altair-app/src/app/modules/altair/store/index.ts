import { InjectionToken } from '@angular/core';
import {
  combineReducers,
  Action,
  ActionReducer,
  ActionReducerMap,
  MetaReducer,
} from '@ngrx/store';

import { environment } from '../../../../environments/environment';

import * as fromLayout from './layout/layout.reducer';
import * as fromQuery from './query/query.reducer';
import * as fromHeaders from './headers/headers.reducer';
import * as fromVariables from './variables/variables.reducer';
import * as fromDialogs from './dialogs/dialogs.reducer';
import * as fromGqlSchema from './gql-schema/gql-schema.reducer';
import * as fromDocs from './docs/docs.reducer';
import * as fromWindows from './windows/windows.reducer';
import * as fromHistory from './history/history.reducer';
import * as fromWindowsMeta from './windows-meta/windows-meta.reducer';
import * as fromSettings from './settings/settings.reducer';
import * as fromDonation from './donation/donation.reducer';
import * as fromCollection from './collection/collection.reducer';
import * as fromEnvironments from './environments/environments.reducer';
import * as fromStream from './stream/stream.reducer';
import * as fromPreRequest from './pre-request/pre-request.reducer';
import * as fromPostRequest from './post-request/post-request.reducer';
import * as fromLocal from './local/local.reducer';
import * as fromAccount from './account/account.reducer';
import { debug } from '../utils/logger';
import { asyncStorageSync } from './async-storage-sync';
import { localStorageSyncConfig } from './local-storage-sync-config';
import { RootState } from 'altair-graphql-core/build/types/state/state.interfaces';
import { AllActions } from './action';

export const getPerWindowReducer = () => {
  const perWindowReducers = {
    layout: fromLayout.layoutReducer,
    query: fromQuery.queryReducer,
    headers: fromHeaders.headerReducer,
    variables: fromVariables.variableReducer,
    dialogs: fromDialogs.dialogReducer,
    schema: fromGqlSchema.gqlSchemaReducer,
    docs: fromDocs.docsReducer,
    history: fromHistory.historyReducer,
    stream: fromStream.streamReducer,
    preRequest: fromPreRequest.preRequestReducer,
    postRequest: fromPostRequest.postRequestReducer,
    windowId: (_ = '') => _,
  };

  return perWindowReducers;
};

// Meta reducer to log actions
export function log(
  _reducer: ActionReducer<any, AllActions>
): ActionReducer<any, AllActions> {
  return (state: RootState, action: AllActions) => {
    if (!environment.production || (window as any).__ENABLE_DEBUG_MODE__) {
      debug.log(action.type, action);
    }
    (window as any).__LAST_ACTION__ = (window as any).__LAST_ACTION__ || [];
    (window as any).__LAST_ACTION__.push(action.type);
    if (environment.production && (window as any).__LAST_ACTION__.length > 10) {
      (window as any).__LAST_ACTION__.shift();
    }

    return _reducer(state, action);
  };
}

export function asyncStorageSyncReducer(
  _reducer: ActionReducer<RootState, AllActions>
): ActionReducer<RootState, AllActions> {
  return asyncStorageSync(localStorageSyncConfig)(_reducer);
}

export const metaReducers: MetaReducer<RootState, AllActions>[] = [
  asyncStorageSyncReducer,
  // !environment.production ? storeFreeze : null,
  log,
];

export const getReducer = (): ActionReducerMap<RootState, AllActions> => {
  return {
    windows: fromWindows.windows(combineReducers(getPerWindowReducer())),
    windowsMeta: fromWindowsMeta.windowsMetaReducer,
    settings: fromSettings.settingsReducer,
    donation: fromDonation.donationReducer,
    collection: fromCollection.collectionReducer,
    environments: fromEnvironments.environmentsReducer,
    local: fromLocal.localReducer,
    account: fromAccount.accountReducer,
  };
};

export const reducerToken = new InjectionToken<ActionReducerMap<RootState>>(
  'Registered Reducers'
);

export const reducerProvider = [
  { provide: reducerToken, useValue: getReducer() },
];

export const selectWindowState = (windowId: string) => (state: RootState) => {
  return state.windows[windowId];
};

export * from './query/selectors';
export * from './docs/selectors';
export * from './headers/selectors';
export * from './variables/selectors';
export * from './layout/selectors';
export * from './gql-schema/selectors';
export * from './collection/selectors';
export * from './pre-request/selectors';
export * from './post-request/selectors';
export * from './stream/selectors';
export * from './local/selectors';
