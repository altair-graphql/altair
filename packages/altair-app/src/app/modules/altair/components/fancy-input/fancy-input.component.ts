import { Component, OnInit, ViewChild, ElementRef, Input, Output, EventEmitter, forwardRef } from '@angular/core';
import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
import { Store } from '@ngrx/store';
import { EnvironmentService } from '../../services';
import { debug } from '../../utils/logger';
import { distinct, distinctUntilChanged, map } from 'rxjs/operators';
import { RootState } from 'altair-graphql-core/build/types/state/state.interfaces';
import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';

// import { VARIABLE_REGEX } from '../../services/environment/environment.service';
// TODO: Duplicating for now after changing to use lookahead in environment service variable regex
export const VARIABLE_REGEX = /{{\s*([\w\.]+)\s*}}/g;
interface BoundaryMarker {
  index: number;
  type: 'start' | 'stop';
  className?: string;
}

export interface HighlightSection {
  content: string;
  type?: string;
}

@UntilDestroy({ checkProperties: true })
@Component({
  selector: 'app-fancy-input',
  templateUrl: './fancy-input.component.html',
  styleUrls: ['./fancy-input.component.scss'],
  preserveWhitespaces: false,
  providers: [
    {
      provide: NG_VALUE_ACCESSOR,
      useExisting: forwardRef(() => FancyInputComponent),
      multi: true,
    }
  ],
})
export class FancyInputComponent implements ControlValueAccessor, OnInit {

  // get accessor
  get value(): any {
    return this.innerValue;
};

  // set accessor including call the onchange callback
  set value(v: any) {
    if (v !== this.innerValue) {
      this.innerValue = v;
      this.onChangeCallback(v);
    }
  }

  // @Input() value = '';
  @Input() placeholder = '';
  @Output() blurChange = new EventEmitter();
  @Output() submitChange = new EventEmitter();

  @ViewChild('fancyInputEl', { static: true }) fancyInputEl: ElementRef;
  @ViewChild('fancyInputHighlightsEl',  { static: true }) fancyInputHighlightsEl: ElementRef;


  highlightData = {
    sections: [] as HighlightSection[]
  };

  private innerValue = '';

  private activeEnvironment = {};

  constructor(
    private store: Store<RootState>,
    private environmentService: EnvironmentService,
  ) {
    store.pipe(
      map(data => data.environments),
      distinctUntilChanged(),
      untilDestroyed(this)
    ).subscribe({
      next: (data) => {
        // get active environment
        this.activeEnvironment = environmentService.getActiveEnvironment();
      },
    });
  }

  // From ControlValueAccessor interface
  writeValue(value: any) {
    if (value !== this.innerValue) {
        this.innerValue = value;
    }
  }

  // From ControlValueAccessor interface
  registerOnChange(fn: any) {
      this.onChangeCallback = fn;
  }

  // From ControlValueAccessor interface
  registerOnTouched(fn: any) {
      this.onTouchedCallback = fn;
  }

  ngOnInit() {
    if (this.detectBrowser() === 'firefox') {
      this.fixFirefox();
    }
    const handleInputTimeout = setTimeout(() => {
      this.handleInput();
      clearTimeout(handleInputTimeout);
    }, 1);
  }

  handleInput() {
    const val = this.fancyInputEl.nativeElement.value;
    const ranges = this.getRanges(val, VARIABLE_REGEX);
    const unstaggeredRanges = this.removeStaggeredRanges(ranges);
    const boundaries = this.getBoundaries(unstaggeredRanges);
    this.highlightData.sections = this.generateHighlightSections(val, boundaries);
    this.updateHighlighterScroll();
  }
  handleScroll() {
    debug.log('scrolling input');
  }
  handleKeydown() {
    this.updateHighlighterScroll();
  }
  handleBlur() {
    this.updateHighlighterScroll();

    // Set touched on blur
    this.onTouchedCallback();
    this.blurChange.next(this.innerValue);
  }
  handleSubmit() {
    this.submitChange.next(this.innerValue);
  }
  blockContainerScroll() {}
  detectBrowser() {
    const ua = window.navigator.userAgent.toLowerCase();
    if (ua.indexOf('firefox') !== -1) {
      return 'firefox';
    } else if (!!ua.match(/msie|trident\/7|edge/)) {
      return 'ie';
    } else if (!!ua.match(/ipad|iphone|ipod/) && ua.indexOf('windows phone') === -1) {
      // Windows Phone flags itself as "like iPhone", thus the extra check
      return 'ios';
    } else {
      return 'other';
    }
  }
  fixFirefox() {}
  getRanges(val: string, highlight: RegExp) {
    const ranges: any[] = [];
    let match;
    while (match = highlight.exec(val), match !== null) {
      ranges.push([match.index, match.index + match[0].length]);
      if (!highlight.global) {
        // non-global regexes do not increase lastIndex, causing an infinite loop,
        // but we can just break manually after the first match
        break;
      }
    }
    return ranges;
  }

  // Prevent overlapping ranges
  removeStaggeredRanges(ranges: any) {
    const unstaggeredRanges: any[] = [];
    ranges.forEach((range: any) => {
      const isStaggered = unstaggeredRanges.some((unstaggeredRange) => {
        const isStartInside = range[0] > unstaggeredRange[0] && range[0] < unstaggeredRange[1];
        const isStopInside = range[1] > unstaggeredRange[0] && range[1] < unstaggeredRange[1];
        return isStartInside !== isStopInside; // xor
      });
      if (!isStaggered) {
        unstaggeredRanges.push(range);
      }
    });
    return unstaggeredRanges;
  }

  getBoundaries(ranges: any[]) {
    const boundaries: BoundaryMarker[] = [];
    ranges.forEach((range) => {
      boundaries.push({
        type: 'start',
        index: range[0],
        className: range.className
      });
      boundaries.push({
        type: 'stop',
        index: range[1]
      });
    });

    this.sortBoundaries(boundaries);
    return boundaries;
  }

  sortBoundaries(boundaries: BoundaryMarker[]) {
    // backwards sort (since marks are inserted right to left)
    boundaries.sort(function(a, b) {
      if (a.index !== b.index) {
        return a.index - b.index;
      } else if (a.type === 'start' && b.type === 'stop') {
        return 1;
      } else if (a.type === 'stop' && b.type === 'start') {
        return -1;
      } else {
        return 0;
      }
    });
  }
  generateHighlightSections(val: string, boundaries: BoundaryMarker[]) {
    const sections: HighlightSection[] = [];
    let lastBoundary = {
      index: 0,
      type: ''
    };

    boundaries.forEach(boundary => {
      sections.push({
        content: val.substring(lastBoundary.index, boundary.index),
        type: lastBoundary.type === 'start' && boundary.type === 'stop' ? 'mark' : undefined,
      });
      lastBoundary = boundary;
    });
    sections.push({
      content: val.substring(lastBoundary.index)
    });

    return sections;
  }

  updateHighlighterScroll() {
    const updateHighlighterScrollTimeout = setTimeout(() => {
      this.fancyInputHighlightsEl.nativeElement.scrollLeft = this.fancyInputEl.nativeElement.scrollLeft;
      this.fancyInputHighlightsEl.nativeElement.scrollTop = this.fancyInputEl.nativeElement.scrollTop;
      this.fancyInputHighlightsEl.nativeElement.height = this.fancyInputEl.nativeElement.height;
      clearTimeout(updateHighlighterScrollTimeout);
    }, 10);
    // highlighter.scrollLeft = input.scrollLeft
    // highlighter.scrollTop = input.scrollTop
    // highlighter.height = input.height
  }

  trackByIndex(index: number) {
    return index;
  }
  private onTouchedCallback: () => void = () => {};
  private onChangeCallback: (_: any) => void = () => {};

}
