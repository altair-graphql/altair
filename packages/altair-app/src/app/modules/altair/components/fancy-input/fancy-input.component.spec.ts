import { expect, it, beforeEach, describe } from '@jest/globals';
import {
  ComponentFixture,
  TestBed,
  tick,
  fakeAsync,
  waitForAsync,
} from '@angular/core/testing';

import { FancyInputComponent } from './fancy-input.component';
import { FormsModule } from '@angular/forms';
import { SharedModule } from '../../modules/shared/shared.module';
import { MockModule } from 'ng-mocks';
import { Store } from '@ngrx/store';
import { mock, mockStoreFactory } from '../../../../../testing';
import { FancyInputMarkerComponent } from '../fancy-input-marker/fancy-input-marker.component';
import { EnvironmentService } from '../../services';

describe('FancyInputComponent', () => {
  let component: FancyInputComponent;
  let fixture: ComponentFixture<FancyInputComponent>;

  beforeEach(
    waitForAsync(() => {
      TestBed.configureTestingModule({
        declarations: [FancyInputComponent, FancyInputMarkerComponent],
        imports: [FormsModule, SharedModule],
        providers: [
          {
            provide: Store,
            useValue: mockStoreFactory(),
          },
          {
            provide: EnvironmentService,
            useValue: mock<EnvironmentService>({
              getActiveEnvironment: () => ({}),
              hydrate: () => '',
            }),
          },
        ],
      }).compileComponents();
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(FancyInputComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should set component value when input value changes', async () => {
    const input = fixture.nativeElement.querySelector('input');
    input.value = 'some text';
    input.dispatchEvent(new Event('input'));
    await fixture.whenStable();
    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot();
    expect(component.value).toBe('some text');
  });

  it('should render the highlights correctly based on the matched variables in text', async () => {
    const input = fixture.nativeElement.querySelector('input');
    input.value = 'some text {{variable1}} then {{variable2}} is next';
    input.dispatchEvent(new Event('input'));
    await fixture.whenStable();
    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should render mark element containing the matched variables in text', async () => {
    const input = fixture.nativeElement.querySelector('input');
    input.value = 'some text {{variable1}} then {{variable2}} is next';
    input.dispatchEvent(new Event('input'));
    await fixture.whenStable();
    fixture.detectChanges();

    const marks = fixture.nativeElement.querySelectorAll('mark');

    expect(marks.length).toBe(2);
    expect(marks[0].textContent).toBe('{{variable1}}');
    expect(marks[1].textContent).toBe('{{variable2}}');
  });
});
