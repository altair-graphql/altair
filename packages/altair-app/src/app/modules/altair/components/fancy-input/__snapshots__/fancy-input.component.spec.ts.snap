// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FancyInputComponent should render the highlights correctly based on the matched variables in text 1`] = `
<div
  id="root2"
>
  <div
    class="fancy-input-container"
  >
    <input
      class="fancy-input-content fancy-input-element ng-untouched ng-valid ng-dirty"
      placeholder=""
      type="text"
    />
    <div
      class="fancy-input-backdrop"
    >
      <div
        class="fancy-input-content fancy-input-highlights"
      >
        <span>
          some text 
        </span>
        
        
        
        <app-fancy-input-marker>
          <mark
            class="fancy-input-marker__invalid"
          >
            {{variable1}}
          </mark>
          
        </app-fancy-input-marker>
        
        <span>
           then 
        </span>
        
        
        
        <app-fancy-input-marker>
          <mark
            class="fancy-input-marker__invalid"
          >
            {{variable2}}
          </mark>
          
        </app-fancy-input-marker>
        
        <span>
           is next
        </span>
        
        
        
      </div>
    </div>
  </div>
</div>
`;

exports[`FancyInputComponent should set component value when input value changes 1`] = `
<div
  id="root1"
>
  <div
    class="fancy-input-container"
  >
    <input
      class="fancy-input-content fancy-input-element ng-untouched ng-valid ng-dirty"
      placeholder=""
      type="text"
    />
    <div
      class="fancy-input-backdrop"
    >
      <div
        class="fancy-input-content fancy-input-highlights"
      >
        <span>
          some text
        </span>
        
        
        
      </div>
    </div>
  </div>
</div>
`;
